pkg/utils/experimental/experimental.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package experimental
+
+import (
+	"os"
+)
+
+var experimental bool
+
+const EnvName = "IG_EXPERIMENTAL"
+
+func init() {
+	exp := os.Getenv(EnvName)
+	if exp == "true" {
+		experimental = true
+	}
+}
+
+func Enabled() bool {
+	return experimental
```

Comment:
AFAIU the variable `experimental` should be fine, as dependencies are always initialized first. So whenever an `init()` function would use `experimental.Enabled()`, it would also lead to the `experimental` packages' initialization.

"If a package has imports, the imported packages are initialized before initializing the package itself. If multiple packages import a package, the imported package will be initialized only once. The importing of packages, by construction, guarantees that there can be no cyclic initialization dependencies." [Source](https://go.dev/ref/spec#Order_of_evaluation)

Reply:
Good catch! I used sync.Once that IMO is cleaner than parsing the env variable each time.

Reply:
Ah, so I think I can go back to my original idea.

Reply:
The logic changed a lot since we first discussed it. I implemented this by doing a lazy initialization and I want to keep it that way to be sure it's called at the right time. This is not part of any hot path so it should be all fine.